### PR TITLE
Do not use enums for API data: there may be a mismatch

### PR DESCRIPTION
**Pull request by blootsvoets**
This avoids a situation where a mismatch does not import the source types.

### CODE REVIEW

File: src/main/java/org/radarbase/management/config/SourceTypeLoader.java [hunk 1]
```diff
@@ -134,21 +134,29 @@ public void saveSourceTypesFromCatalogServer(List<CatalogSourceType> catalogSour
...
+                                    + "_" + sourceType.getCatalogVersion()
+                                    + "_" + sourceData.getSourceDataType());
+                            sourceData.sourceType(sourceType);
+                            sourceDataRepository.save(sourceData);
+                        } catch (RuntimeException ex) {
+                            log.error("Failed to import source data {}", catalogSourceData, ex);
+                        }
+                    }
+                } catch (RuntimeException ex) {
+                    log.error("Failed to import source type {}", sourceType, ex);
```
Comment (baixiac) [inline src/main/java/org/radarbase/management/config/SourceTypeLoader.java:159]: Looks like `catch`es are only used for logging here. Not sure about the impact given the exceptions are swallowed.

**Review by baixiac — COMMENTED** [review]
LGTM. One question about the error handling.

**Review by nivemaham — APPROVED** [review]

File: src/main/java/org/radarbase/management/config/SourceTypeLoader.java (hunk 1 above)
Comment (blootsvoets) [reply src/main/java/org/radarbase/management/config/SourceTypeLoader.java:159]: The exception swallowing is intentional. The previous behaviour was that exceptions were swallowed one level up, but then registration just stopped completely at the malformed source type.

Comment (baixiac) [reply src/main/java/org/radarbase/management/config/SourceTypeLoader.java:159]: 👍

**Pull request merged**